### PR TITLE
Add Flatten layer

### PR DESCRIPTION
**Pull request by Vimalakirti**

### CODE REVIEW

File: src/layer/flatten.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  let permutation = ArrayD::from_shape_fn(input_shape, |index| Some(index));
+  let permutation = permutation.view().into_shape(&output_shape.clone()[..]).unwrap().to_owned();
+  let padded_permutation = util::pad_to_pow_of_two(&permutation, &None);
+
+  (padded_permutation, output_shape)
+}
+
+// https://onnx.ai/onnx/operators/onnx__Flatten.html
+// Flattens the input tensor into a 2D matrix.
+// If input tensor has shape (d_0, d_1, … d_n) then the output will have shape (d_0 X d_1 … d_(axis-1), d_axis X d_(axis+1) … X dn).
```
Comment (ddkang) [inline src/layer/flatten.rs]: I don't think this parses

Comment (liliatangxy) [reply src/layer/flatten.rs]: I think we can add 'X's before and after the '...'s

**Review by liliatangxy — APPROVED** [review]
lgtm after fixing the comment

File: src/layer/flatten.rs (hunk 1 above)
Comment (Vimalakirti) [reply src/layer/flatten.rs]: it was rewritten

**Review by ddkang — APPROVED** [review]

**Pull request merged**